phylo/src/evolutionary_models/mod.rs

```diff
@@ -61,34 +79,55 @@ impl Display for ProteinModelType {
     }
 }
 
-pub trait EvolutionaryModelParameters<T> {
-    fn new(model_type: &T, params: &[f64]) -> Result<Self>
+impl ProteinModelType {
+    pub fn get_model_type(model_name: &str) -> Self {
```

Comment:
nit: you either wanna name this `from()`, or better use `std::convert` https://doc.rust-lang.org/std/convert/trait.From.html